src/validators/conditional-answered-validator.ts

```diff
@@ -0,0 +1,16 @@
+import { type FormFieldValidator, type FormField } from '../types';
+
+export const conditionalAnsweredValidator: FormFieldValidator = {
+  validate: function (field: FormField, value: unknown, config: Record<string, any>) {
+    const { referenceQuestionId, referenceQuestionAnswers, values, fields, message } = config;
+
+    const referencedField = fields.find((field) => field.id === referenceQuestionId);
+    const referencedFieldValue = values[referencedField.id] || referencedField.meta?.submission?.newValue?.value;
```

Comment:
The second part of the expressions presupposes the underlying field to be capturing an obs. I think referencing the `values` object should suffice.